### PR TITLE
Add tests for parsing of empty queries and remove redundant checks

### PR DESCRIPTION
**Pull request by robertknight**
This commit started out as a fix for a typecheck error in TypeScript 4.9.3 in `search-filter.js` that led to a discovery of missing tests for handling of empty inputs and redundant checks for empty inputs in the implementation.

Add tests for empty queries and clean up the implementation. I suggest reviewing this with whitespace changes hidden.

There are no substantial changes here, though I did change the edge-case handling so that the functions return the same result when the input is an empty pair of quotes (`""`) as they do when the input is an empty string. Due to the way downstream code works, this change doesn't have an obvious user-facing impact. Searching for whitespace-only strings or empty quotes (`""`) will still end up showing all annotations and replies that were fetched from the backend. I think that's undesirable behavior, but fixing it is out-of-scope for this PR.

### CODE REVIEW

File: src/sidebar/util/test/search-filter-test.js [hunk 1]
```diff
@@ -185,7 +224,7 @@ describe('sidebar/util/search-filter', () => {
     });
 
     it('filters to a focused user', () => {
-      const filter = searchFilter.generateFacetedFilter(null, {
+      const filter = searchFilter.generateFacetedFilter('', {
```
Comment (robertknight) [inline src/sidebar/util/test/search-filter-test.js:227]: Per the type definition of `generateFacetedFilter`, the first argument is required to be a string. `null` happened to work previously. App code always passes a (possibly empty) string as the first arg.

**Review by lyzadanger — APPROVED** [review]
👍🏻 Thanks for this improvement.
